### PR TITLE
KAFKA-16139: Fix StreamsUpgradeTest

### PR DESCRIPTION
**Pull request by cadonna**
Adds version 3.5 to the possible values for config upgrade_from.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cadonna** [comment]
@mjsax Could you have a look since you reviewed also the PR that added upgrade tests to version 3.5?

**Comment by cadonna** [comment]
I started Streams system tests here: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/6034/

**Comment by mjsax** [comment]
Is this fix sufficient? Don't we not also need to add `3.6` as "upgrade_from" version?

**Review by mjsax — APPROVED** [review]

**Comment by mjsax** [comment]
I just checked the test failures on https://github.com/apache/kafka/pull/15151 (cf last comment) and it failed because `3.6` is missing.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `3.7` and `3.6` branches.

**Comment by mjsax** [comment]
Follow-up PR to add `3.6`: https://github.com/apache/kafka/pull/15207

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -71,6 +71,7 @@
 import static org.apache.kafka.common.config.ConfigDef.Range.between;
 import static org.apache.kafka.common.config.ConfigDef.ValidString.in;
 import static org.apache.kafka.common.config.ConfigDef.parseType;
+import static org.apache.kafka.streams.internals.UpgradeFromValues.UPGRADE_FROM_35;
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:74]: After merging and doing the follow up PR to add `3.6`, I realized that using this import is not what we want to do.

Call for review (follow up cleanup PR): https://github.com/apache/kafka/pull/15208

**Pull request merged**